Main.py

```diff
@@ -541,17 +543,25 @@ def create_playthrough(spoiler):
     # Regenerate the spheres as we might not reach places the same way anymore.
     playthrough.reset()  # playthrough state has no items, okay to reuse sphere 0 cache
     collection_spheres = []
+    entrance_spheres = []
+    previously_accessed_entrances = []
     while True:
         # Not collecting while the generator runs means we only get one sphere at a time
         # Otherwise, an item we collect could influence later item collection in the same sphere
         collected = list(playthrough.iter_reachable_locations(required_locations))
+        accessed_entrances = [entrance for world in worlds for entrance in world.get_shuffled_entrances() 
+                                if entrance not in previously_accessed_entrances and entrance.primary and playthrough.state_list[world.id].can_reach(entrance)]
         if not collected: break
         for location in collected:
             # Collect the item for the state world it is for
             state_list[location.item.world.id].collect(location.item)
         collection_spheres.append(collected)
+        entrance_spheres.append(accessed_entrances)
+        previously_accessed_entrances.extend(accessed_entrances)
     logger.info('Collected %d final spheres', len(collection_spheres))
 
     # Then we can finally output our playthrough
     spoiler.playthrough = OrderedDict((str(i + 1), {location: location.item for location in sphere}) for i, sphere in enumerate(collection_spheres))
 
+    if worlds[0].entrance_shuffle != 'off':
+        spoiler.entrance_playthrough = OrderedDict((str(i + 1), [entrance for entrance in sphere]) for i, sphere in enumerate(entrance_spheres))
```

Comment:
I also think it could be better to include it in the normal playthrough, but since it doesn't only include useful entrances, I left it in a separate playthrough for now. 
I wanna try to have it only include entrances actually useful to items collected in the playthrough, but I'm not sure how to do that without losing too much performance.
Once it works this way, it will probably make more sense to include it in the item playthrough, sorted last as you mentioned.